### ISSUE TITLE
Binance spot fetch my trades broken

### ISSUE THREAD

**Issue by acodemics** [ccxt/ccxt#16926]
### Operating System

Linux

### CCXT Version

2.8.24

### Description

Hello,

We recently had two problems fetching own trades from Binance spot margin mode.

The first is that setting `defaultType: margin` no longer works in the CCXT options, it still defaults to spot when requesting trades. I added `marginMode: True` as a work around and now I can see it retrieving trades.

The second issue is that it seems like all of the found trades get filtered out and not returned by the `fetch_my_trades` method. I can see them in the debug log but not returned by the method.

Is there anything more we need to change in our CCXT constructor to get this to work?

Thank you.

### Code

```
  

```

**Comment by carlosmiei** [comment]
Hello @acodemics, thanks for reporting it, I will take a look asap.

Just a side note, binance was recently updated and unified, so `defaultType:margin` is no longer advisable. 
Also, you're providing an invalid value for `marginMode`, it should be either `isolated` or `cross`.

Nonetheless, I will try to reproduce the issue reported and fix it if needed!

**Comment by acodemics** [comment]
Just a note that this is in the Python library - is it possible to get the same fix there?

**Comment by carlosmiei** [comment]
> Just a note that this is in the Python library - is it possible to get the same fix there?

@acodemics The fix is applied automatically in Python and PHP because our JS code is transpiled.

**Comment by acodemics** [comment]
Understood, thanks for your help!

**Comment by acodemics** [comment]
We still have a bug where fetch_my_trades is filtering out all margin trades.

We are setting the symbol to be `BTC/USDT` which gets the trades from Binance but then in parse trades it seems to set the symbol field to `BTC/USDT:USDT` in each array entry then they all get filtered out.

`ccxt.binance(
            {
                "apiKey": ...,
                "secret": ...,
                "options": {
                    "defaultType": "spot",
                    "marginMode": "cross",
                    "recvWindow": 60 * 1000  # 60 seconds
                }
            }
        )`

Here is our set up for Binance spot margin, is this correct?

Thank you for your help.

**Comment by carlosmiei** [comment]
> We still have a bug where fetch_my_trades is filtering out all margin trades.
> 
> We are setting the symbol to be `BTC/USDT` which gets the trades from Binance but then in parse trades it seems to set the symbol field to `BTC/USDT:USDT` in each array entry then they all get filtered out.
> 
> `ccxt.binance( { "apiKey": ..., "secret": ..., "options": { "defaultType": "spot", "marginMode": "cross", "recvWindow": 60 * 1000 # 60 seconds } } )`
> 
> Here is our set up for Binance spot margin, is this correct?
> 
> Thank you for your help.

@acodemics Yes seems like the right config, what's the current ccxt version you're using?

**Comment by acodemics** [comment]
We upgraded to the latest `2.8.31` after the last patch.

**Comment by carlosmiei** [comment]
> We upgraded to the latest `2.8.31` after the last patch.

@acodemics Can you please share a verbose log?

**Comment by acodemics** [comment]
[out.log](https://github.com/ccxt/ccxt/files/10813136/out.log)

Attached a stripped down log of what happens - when using the symbol `BTCUSDT` and spot margin we correctly get a list of trades from the Binance API but then internally in CCXT `parse_trades` the trade symbol gets changed to `BTC/USDT:USDT` which is then filtered out by `filter_by_symbol_since_limit`.
